tests/test_merge.py

```diff
@@ -199,7 +202,9 @@ def test_merge_destination_2(tmp_path):
         with rasterio.open(tmp_path.joinpath("test.tif")) as dst:
             result = dst.read()
             assert result.shape == (3, 719, 792)
-            assert numpy.allclose(data.mean(), result[:, :-1, :-1].mean())
+            assert numpy.allclose(
+                data[data != 0].mean(), result[result != 0].mean(), rtol=1e-3
```

Comment:
Maybe implement a difference hash?

FWIW, the result looks fine in QGIS:

![Untitled](https://github.com/user-attachments/assets/3aa29109-59c6-4e2b-b53c-e577ceda9454)